Content.Server/Chat/Systems/LastMessageBeforeDeathSystem.cs

```diff
@@ -0,0 +1,151 @@
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Adds a message for a player to the internal data structure.
+        ///     Truncates the message if it exceeds the maximum length.
+        /// </summary>
+        /// <param name="source">The entity UID of the message source.</param>
+        /// <param name="player">The player's username.</param>
+        /// <param name="message">The message content.</param>
+        /// <param name="playerName">The name of the player's character.</param>
+        public async void AddMessage(EntityUid source, string player, string message, string playerName)
+        {
+
+            var actorComponent = _entManager.GetComponentOrNull<ActorComponent>(source);
+            if (actorComponent?.PlayerSession?.UserId != null)
+            {
+                _userNetId = actorComponent.PlayerSession.UserId;
+            }
+            else
+            {
+                _userNetId = null;
+            }
+
+            if (message.Length > MaxICLength)
+            {
+                // if the message is bigger than the message length limit, we make it cut off at a random iterval.
+                // Example: Someone was giving a speech and got blown to bits.
+                var randomLength = _random.Next(1, MaxICLength);
+                message = message[..randomLength] + "-";
+            }
+
+            var currentTime = _gameTicker.RoundDuration();
+            var truncatedTime = $"{currentTime.Hours:D2}:{currentTime.Minutes:D2}:{currentTime.Seconds:D2}";
+            var formattedMessage = $"[{truncatedTime}] {playerName}: {message}";
+
+            if (!_playerData.Contains(player))
+            {
+                _playerData[player] = new OrderedDictionary();
+            }
+
+            var playerMessages = _playerData[player] as OrderedDictionary ?? throw new InvalidOperationException("Player messages dictionary is not initialized.");
+            playerMessages[playerName] = formattedMessage;
+            playerMessages["EntityUid"] = source;
+            playerMessages["NetUserId"] = _userNetId;
+        }
+
+        /// <summary>
+        ///     Sends all collected messages through the webhook at the end of the round.
+        ///     Skips messages from banned users.
+        /// </summary>
+        /// <param name="webhookId">The identifier of the webhook to send messages to.</param>
+        public async void OnRoundEnd(WebhookIdentifier? webhookId)
+        {
+            if (!webhookId.HasValue)
+                return;
+
+            var id = webhookId.Value;
+            StringBuilder allMessagesString = new StringBuilder();
+            foreach (DictionaryEntry player in _playerData)
+            {
+                var playerMessages = player.Value as OrderedDictionary;
+                if (playerMessages != null)
+                {
+                    if (playerMessages["EntityUid"] is EntityUid playerUid && _mobStateSystem.IsDead(playerUid))
```

Comment:
This won't log people that have been gibbed, because `IsDead` will fail for those.

Reply:
Yeah... I am not sure how to go about this... check if an entity still exists and if not consider dead?

Reply:
Probably.